nilearn/glm/_base.py

```diff
@@ -227,9 +295,178 @@ def generate_report(
             alpha=alpha,
             cluster_threshold=cluster_threshold,
             height_control=height_control,
+            two_sided=two_sided,
             min_distance=min_distance,
             plot_type=plot_type,
             cut_coords=cut_coords,
             display_mode=display_mode,
             report_dims=report_dims,
         )
+
+    def _generate_filenames_output(
+        self, prefix, contrasts, contrast_types, out_dir
+    ):
+        """Generate output filenames for a series of contrasts.
+
+        Store the name of the output files in the model.
+
+        See nilearn.interfaces.bids.save_glm_to_bids for more details.
+
+        Parameters
+        ----------
+        prefix : :obj:`str`
+            String to prepend to generated filenames.
+            If a string is provided, '_' will be added to the end.
+
+        contrasts : :obj:`str` or array of shape (n_col) or :obj:`list` \
+                of (:obj:`str` or array of shape (n_col)) or :obj:`dict`
+                Contrast definitions.
+
+        contrast_types ::obj:`dict` of :obj:`str`
+            An optional dictionary mapping some
+            or all of the :term:`contrast` names to
+            specific contrast types ('t' or 'F').
+
+        out_dir : :obj:`str` or :obj:`pathlib.Path`
+            Output directory for files.
+        """
+        check_is_fitted(self)
+
+        if not isinstance(prefix, str):
+            prefix = ""
+        if prefix and not prefix.endswith("_"):
+            prefix += "_"
+
+        contrasts = coerce_to_dict(contrasts)
+        for k, v in contrasts.items():
+            if not isinstance(k, str):
+                raise ValueError(
+                    f"contrast names must be strings, not {type(k)}"
+                )
+
+            if not isinstance(v, (str, np.ndarray, list)):
+                raise ValueError(
+                    "contrast definitions must be strings or array_likes, "
+                    f"not {type(v)}"
+                )
+
+        if self.__str__() == "Second Level Model":
```

Comment:
Could "Second Level Model" be defined as class constant in `nilearn.glm.second_level.second_level.SecondLevelModel` and comparison use that?

Reply:
SecondLevelModel can have subdirectory property, and design_matrices property that returns the correct result for itself to avoid if conditions (if these are checked elsewhere as well).